server/channels/app/scheduled_post_job.go

```diff
@@ -0,0 +1,272 @@
+			mlog.String("user_id", scheduledPost.UserId),
+			mlog.String("channel_id", scheduledPost.ChannelId),
+			mlog.Err(err),
+		)
+
+		return scheduledPost, appErr
+	}
+
+	if !canPost {
+		rctx.Logger().Warn(
+			"App.processScheduledPostBatch: skipping posting a scheduled post as `can post` check failed",
+			mlog.String("scheduled_post_id", scheduledPost.Id),
+			mlog.String("user_id", scheduledPost.UserId),
+			mlog.String("channel_id", scheduledPost.ChannelId),
+			mlog.String("error_code", scheduledPost.ErrorCode),
+		)
+
+		return scheduledPost, appErr
+	}
+
+	post, err := scheduledPost.ToPost()
+	if err != nil {
+		rctx.Logger().Error(
+			"App.processScheduledPostBatch: failed to convert scheduled post job to a post",
+			mlog.String("scheduled_post_id", scheduledPost.Id),
+			mlog.Err(err),
+		)
+
+		return scheduledPost, appErr
+	}
+
+	_, appErr = a.CreatePost(rctx, post, channel, true, false)
+	if appErr != nil {
+		rctx.Logger().Error(
+			"App.processScheduledPostBatch: failed to post scheduled post",
+			mlog.String("scheduled_post_id", scheduledPost.Id),
+			mlog.String("channel_id", scheduledPost.ChannelId),
+			mlog.Err(appErr),
+		)
+
+		return scheduledPost, appErr
+	}
+
+	return scheduledPost, nil
+}
+
+// canPostScheduledPost checks whether the scheduled post be created based on permissions and other checks.
+func (a *App) canPostScheduledPost(rctx request.CTX, scheduledPost *model.ScheduledPost, channel *model.Channel) (*model.ScheduledPost, bool, error) {
+	user, appErr := a.GetUser(scheduledPost.UserId)
+	if appErr != nil {
+		if appErr.Id == MissingAccountError {
+			scheduledPost.ErrorCode = model.ScheduledPostErrorCodeUserDoesNotExist
+			return scheduledPost, false, nil
+		}
+
+		rctx.Logger().Error(
+			"App.canPostScheduledPost: failed to get user from database",
+			mlog.String("user_id", scheduledPost.UserId),
+			mlog.Err(appErr),
+		)
+		scheduledPost.ErrorCode = model.ScheduledPostErrorUnknownError
+		return scheduledPost, false, errors.Wrapf(appErr, "App.canPostScheduledPost: failed to get user from database, userId: %s", scheduledPost.UserId)
+	}
+
+	if user.DeleteAt != 0 {
+		scheduledPost.ErrorCode = model.ScheduledPostErrorCodeUserDeleted
+		return scheduledPost, false, nil
+	}
+
+	if channel.DeleteAt != 0 {
+		scheduledPost.ErrorCode = model.ScheduledPostErrorCodeChannelArchived
+		return scheduledPost, false, nil
+	}
+
+	_, appErr = a.GetChannelMember(rctx, scheduledPost.ChannelId, scheduledPost.UserId)
+	if appErr != nil {
+		if appErr.Id == MissingChannelMemberError {
+			scheduledPost.ErrorCode = model.ScheduledPostErrorNoChannelMember
+			return scheduledPost, false, nil
+		}
+
+		rctx.Logger().Error(
+			"App.canPostScheduledPost: failed to get channel member",
+			mlog.String("user_id", scheduledPost.UserId),
+			mlog.String("channel_id", scheduledPost.ChannelId),
+			mlog.Err(appErr),
+		)
+		scheduledPost.ErrorCode = model.ScheduledPostErrorUnknownError
```

Comment:
I've refactored this so that I'm always setting the right error code in all the palces. So, I now removed that for loop where Iw as adding the missing error codes.